### PR TITLE
Update listing model

### PR DESCRIPTION
**Pull request by kingbrendann**

### CODE REVIEW

**Comment by kingbrendann** [comment]
This is ready

**Comment by kingbrendann** [comment]
Just kidding. Now it's done

File: controllers/listingController.js [hunk 1]
```diff
@@ -79,39 +75,70 @@ var ListingController = {
...
+                imageId: v.image_id,
+                inventoryQuantity: v.inventory_quantity
+            };
+            variants.push(variant);
+        });
+
         listing.title = item.title;
         listing.description = item.body_html;
-        listing.previewImageUrl = item.images[0].src;
+        listing.feedImageUrl = item.images[0].src;
```
Comment (dodgertodd) [inline controllers/listingController.js:127]: There's got to be something that prevents this from happening. Some kind of compilation-like check that doesn't let you run with a reference to a value/function that doesn't exist.

File: models/listing.js [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+// http://mongoosejs.com/docs/subdocs.html
+var productVariantSchema = new mongoose.Schema({
+    vendorId: Number,
+    title: String,
+    sku: String,
+    option1: String,
+    option2: String,
+    option3: String,
+    imageId: Number,
+    inventoryQuantity: Number
```
Comment (dodgertodd) [inline models/listing.js:15]: How do we want the client to know whether to render "sold out" or not? Sum the quantity of all variants?

**Review by dodgertodd — APPROVED** [review]

File: controllers/listingController.js (hunk 1 above)
Comment (kingbrendann) [reply controllers/listingController.js:127]: I think there is. Mongoose has some kind of .valid check we could run I think.

**Pull request merged**